### PR TITLE
GumGum: Add product parameter

### PR DESCRIPTION
**Pull request by gg-natalia**
In these changes, we are starting to pass the product param from Prebid Server to GG Ad Server under the impressions.ext param.
For any questions/comments pls feel free to reach out.
Thanks
Natalia Castiglioni

### CODE REVIEW

File: adapters/gumgum/gumgum.go [hunk 1]
```diff
@@ -176,6 +176,10 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
 		}
 	}
 
+	if gumgumExt.Product != "" {
+		imp.Ext = json.RawMessage(fmt.Sprintf(`{ "product": "%s" }`, gumgumExt.Product))
```
Comment (SyntaxNode) [inline adapters/gumgum/gumgum.go:180]: Please use json.Marshal or jsonutil.Marshal instead. Using a json.RawMessage exposes a json injection attack.

Comment (gg-natalia) [reply adapters/gumgum/gumgum.go]: Changed 👍 ty

File: adapters/gumgum/gumgum.go [hunk 2]
```diff
@@ -176,6 +176,15 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
...
 	}
 
+	if gumgumExt.Product != "" {
+		var err error
+		//imp.Ext, err = json.Marshal(`{ "product": "%s" }`, gumgumExt.Product)
+		imp.Ext, err = json.Marshal(map[string]string{"product": gumgumExt.Product})
+		if err != nil {
+			return nil, err
+		}
+	}
```
Comment (onkarvhanumante) [inline adapters/gumgum/gumgum.go:186]: should update json tests for above change

Comment (gg-natalia) [reply adapters/gumgum/gumgum.go]: Added a couple of valid/invalid params for test cases. Thanks

File: adapters/gumgum/gumgum.go [hunk 3]
```diff
@@ -1,230 +1,234 @@
...
+    return nil
 }
 
 // Builder builds a new instance of the GumGum adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
-	bidder := &GumGumAdapter{
-		URI: config.Endpoint,
-	}
-	return bidder, nil
+    bidder := &GumGumAdapter{
```
[the remark this answers is not in the document]
Comment (onkarvhanumante) [reply adapters/gumgum/gumgum.go]: @gg-natalia  requesting to provide feedback on above comment

[the remark this answers is not in the document]
Comment (gg-natalia) [reply adapters/gumgum/gumgum.go]: Sorry, missed this, changed 👍

**Review by Sonali-More-Xandr — DISMISSED** [review]

File: adapters/gumgum/gumgum.go [hunk 4]
```diff
@@ -176,6 +176,14 @@ func preprocess(imp *openrtb2.Imp) (*openrtb_ext.ExtImpGumGum, error) {
...
 	}
 
+	if gumgumExt.Product != "" {
+		var err error
+		imp.Ext, err = json.Marshal(map[string]string{"product": gumgumExt.Product})
+		if err != nil {
+			return nil, err
+		}
+	}
+
```
Comment (onkarvhanumante) [inline adapters/gumgum/gumgum.go:186]: @gg-natalia  should add /update json framework tests for PR change

Comment (gg-natalia) [reply adapters/gumgum/gumgum.go:186]: Added one file, ty

**Review by onkarvhanumante — APPROVED** [review]

**Comment by gg-natalia** [comment]
@onkarvhanumante @Sonali-More-Xandr  Hey, do you know if anything is pending on my side ?  Thanks!!!

**Review by Sonali-More-Xandr — APPROVED** [review]

**Comment by gg-natalia** [comment]
Hey, any idea on when this going to be merged ? Thanks!

**Pull request merged**